src/components/comment/newComment/CommentForm.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+
+// MUI
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+
+// Redux
+import { connect } from 'react-redux';
+import { getCommentPost } from '../../../redux/actions/dataActions';
+
+const CommentForm = ({ classes = {}, postId, UI, getCommentPost }) => {
+  const [body, setBody] = useState('');
+
+  const handleChange = event => {
+    const { value } = event.target;
+    setBody(value);
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    const commentData = { body };
+    getCommentPost(postId, commentData);
+  };
+
+  useEffect(() => {
+    if (!UI.error.comment && !UI.isLoading) {
+      setBody('');
+    }
+  }, [UI.error, UI.isLoading]);
+  return (
+    <Grid item sm={12} style={{ textAlign: 'center' }}>
+      <form onSubmit={handleSubmit}>
+        <TextField
+          name='body'
+          type='text'
+          label='Comment on post'
+          error={UI.error.comment ? true : false}
+          helperText={UI.error.comment}
```

Comment:
can UI.error ever not exist?

Reply:
By default, it's always empty. It's only ever occupied if there is an error.

Reply:
but can error ===null|undefined? if so this might crash

Reply:
The initial state of error is always established as empty. When errors are cleared , it is re-established as an empty object, so as far as I can tell, it should never be null or undefined.